src/wp-includes/link-template.php

```diff
@@ -2511,7 +2511,12 @@ function get_next_posts_page_link( $max_page = 0 ) {
  * @return string|void The link URL for next posts page if `$display = false`.
  */
 function next_posts( $max_page = 0, $display = true ) {
-	$output = esc_url( get_next_posts_page_link( $max_page ) );
+	$link = get_next_posts_page_link( $max_page );
+
+	$output = '';
+	if ( $link ) {
+		$output = esc_url( $link );
+	}
```

Comment:
```suggestion
	$out    = get_next_posts_page_link( $max_page );
	$output = $link ? esc_url( $link ) : '';
```

Use ternary operator for simplify it.

Reply:
I think either approach is fine. I'm not opinionated about either way. Both ways exist in Core.